examples/apps/contact-collection/src/containerCode.ts

```diff
@@ -32,7 +32,8 @@ class ContactCollectionContainerRuntimeFactoryType extends BaseContainerRuntimeF
     }
 
     protected async containerInitializingFirstTime(runtime: IContainerRuntime) {
-        await runtime.createRootDataStore(ContactCollectionInstantiationFactory.type, contactCollectionId);
+        const dataStore = await runtime.createDataStore(ContactCollectionInstantiationFactory.type);
+        await dataStore.trySetAlias(contactCollectionId);
```

Comment:
Random observations:
- AliasResult enum does not have good comments describing what values mean and how to react to them.
- "Aliasing" is really weird return value. Can we avoid it? Ideal solution is the one where you return previous promise, and thus this state is not a state that client needs to know about it. As of now, I do not know what to do if I get that return value - there is no right path forward.

Reply:
I feel like we need to fix both in order to move forward.

Reply:
I can add docs about the result values. For the second issue, I agree with the idea, however that needs to go in `next` as technically it breaks back-compat.

Reply:
Do we have guidance around what to do on clients that fail to set alias? I am thinking about "createRootDataStore" type scenario where multiple clients race to create and alias a data store and only one of them wins. What should the loser do? Should it throw away the data store it created or copy its contents over, etc.?

I am concerned what happens if the end up using the data store that lost without storing the handle thinking its root. Also, we need to mention that aliased data stores are not GC'd and there is no need to store its handle anywhere.

Reply:
Yes, we do. Maybe it's not articulated well enough in the docs but the idea is to not use the datastore until it is aliased. If the aliasing fails, the datastore should be thrown away (will end up GCed anyway if the handle is lost) and the aliased datastore retrieved as used as the source of truth. If the datastore is used, then the client is responsible for reconciling the data themselves, after aliasing.

> I am concerned what happens if the end up using the data store that lost without storing the handle thinking its root.

Not sure I follow this. If someone creates a datastore, it is nameless. I don't believe there is an assumption that it is root. It's the same situation as of right now, actually, as the aliasing API make you use the existing flow of creating regular datastores. TrySetAlias will attempt to give it a name and unless that attempt is successful, I don't see why one would expect it NOT to be garbage collected if the handle is not stored. I am more concerned with misuse if people actually start storing garbage datastore handles and/or create split-brain problems by working with a datastore which is not aliased alongside the one that is actually aliased.

Reply:
@vladsud updated. Let me know if you have other comments